### ISSUE TITLE
In stability-aware linting mode, stable -> unstable transitions are reported as `#[doc(hidden)]`

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#1672]
This bug occurs when running `cargo-semver-checks` in stability-aware mode (i.e. for the Rust standard library).

If an item is initially stable and becomes unstable, `cargo-semver-checks` correctly identifies that the item is no longer public API. However, the diagnostic text mentions on such lints mentions `#[doc(hidden)]` as the cause, for example:
```
A pub function is now #[doc(hidden)], removing it from the crate's public API.
```

This text is untemplated at the moment, so the fix is a bit non-trivial. We could duplicate lints (oof!) or try to make the text templated and able to mention stability attributes when in stability-aware mode ("either X or Y caused the problem", putting the onus on the user to figure out which even though the tool arguably already could know).
